setup.py

```diff
@@ -859,7 +859,11 @@ def debug_build():
         license="HPND",
         author="Alex Clark (PIL Fork Author)",
         author_email="aclark@python-pillow.org",
-        url="http://python-pillow.org",
+        url="https://python-pillow.org/",
+        project_urls={
+            "Documentation": "https://pillow.readthedocs.io/en/stable/",
```

Comment:
The README uses this more generic URL, let's use that in case we change the default in future:
```suggestion
            "Documentation": "https://pillow.readthedocs.io",
```